src/adapter/http/src/general/dataset_info_handler.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use axum::extract::{Extension, Path};
+use axum::response::Json;
+use database_common_macros::transactional_handler;
+use dill::Catalog;
+use http_common::*;
+use kamu_core::{DatasetRepository, GetDatasetError};
+use opendatafabric::{AccountName, DatasetHandle, DatasetID, DatasetName};
+
+use crate::axum_utils::ensure_authenticated_account;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DatasetInfoResponse {
+    pub id: DatasetID,
+    pub account_name: Option<AccountName>,
```

Comment:
Let's add `account_id` too. Name is good for visualization, but navigating from dataset to an account should be done by IDs that are resistant to renaming.